frontend/src/libs/components/input-file/input-file.tsx

```diff
@@ -0,0 +1,131 @@
+import {
+  ErrorMessage,
+  type FieldValuesFromFieldErrors,
+} from '@hookform/error-message';
+
+import {
+  EMPTY_ARRAY_LENGTH,
+  FIRST_ARRAY_INDEX,
+} from '#libs/constants/constants.js';
+import { IconColor } from '#libs/enums/enums.js';
+import { getValidClassNames } from '#libs/helpers/helpers.js';
+import { useCallback, useFormController } from '#libs/hooks/hooks.js';
+import {
+  type FormControl,
+  type FormFieldErrors,
+  type FormFieldName,
+  type FormFieldPath,
+  type FormFieldValues,
+} from '#libs/types/types.js';
+
+import { Icon } from '../components.js';
+import styles from './styles.module.scss';
+
+type Properties<T extends FormFieldValues> = {
+  control: FormControl<T, null>;
+  errors: FormFieldErrors<T>;
+  name: FormFieldPath<T>;
+  fileName: string | null;
+  label: string;
+  description: string;
+  onChange?: (file: File) => void;
+};
+
+const InputFile = <T extends FormFieldValues>({
+  control,
+  errors,
+  name,
+  fileName,
+  label,
+  description,
+  onChange,
+}: Properties<T>): JSX.Element => {
+  const { field } = useFormController({ name, control });
+
+  const inputErrors = errors[name];
+  const hasError =
+    inputErrors && Object.keys(inputErrors).length > EMPTY_ARRAY_LENGTH;
+  const displayFile = Boolean(fileName) && !hasError;
+
+  const handleFileChange = useCallback(
+    (event_: React.ChangeEvent<HTMLInputElement>) => {
+      if (event_.target.files?.length) {
+        const currentFile = event_.target.files[FIRST_ARRAY_INDEX] as File;
+
+        if (onChange) {
+          onChange(currentFile);
+        }
+
+        const { name, type, size } = currentFile;
+        field.onChange({ name, type, size });
+      }
+    },
+    [field, onChange],
+  );
+
+  const displayError = useCallback(({ message }: { message: string }) => {
+    return message && <span className={styles['error-text']}>{message}</span>;
+  }, []);
+
+  return (
+    <label className={styles['container']}>
+      <span className={styles['label']}>{label}</span>
+      <div
+        className={getValidClassNames(
+          styles['drag-zone'],
+          hasError && styles['error'],
+        )}
+      >
+        <div className={styles['icon-container']}>
+          <Icon name="upload" color={IconColor.BLACK} />
+        </div>
+        <span className={styles['primary-text']}>Drag file or click here</span>
+        <span className={styles['secondary-text']}>{description}</span>
+        <input
+          type="file"
+          className={styles['input']}
+          onChange={handleFileChange}
+        />
+      </div>
+
+      <div>
+        <ErrorMessage
+          errors={errors}
+          name={
+            name.toString() as FormFieldName<
+              FieldValuesFromFieldErrors<FormFieldErrors<T>>
+            >
+          }
+          render={displayError}
```

Comment:
I think it's not, because otherwise we will have our errors as one joined string. As I understand, I should output to user only one error, not all at ones, so I need to separate them to different spans

Reply:
Also have to mention that linter doesn't allow to use arrow functions here, so in order to make custo render I need to create function via `useCallback`